back/app/controllers/web_api/v1/baskets_ideas_controller.rb

```diff
@@ -52,6 +52,51 @@ def update
     end
   end
 
+  def upsert
+    # 1. Create or get basket
+    idea = Idea.find(params[:idea_id])
+    participation_context = ParticipationContextService.new.get_participation_context(idea.project)
+    participation_context_type = participation_context.phase? ? 'Phase' : 'Project'
+
+    basket = Basket.find_or_initialize_by(
+      participation_context: participation_context,
+      participation_context_type: participation_context_type,
+      user: current_user
+    )
+    if basket.new_record?
+      if basket.save
+        SideFxBasketService.new.after_create basket, current_user
+      else
+        render json: { errors: baskets_idea.errors.details }, status: :unprocessable_entity
```

Comment:
Are you sure the baskets_idea will contain the error if the basket fails to save (these are different models)?

Reply:
I'll write a test to check. Think I saw that it was whilst I was writing the tests though.